scylla-cql/src/types/serialize/serialize_row.rs

```diff
@@ -0,0 +1,44 @@
+use std::{any::Any, sync::Arc};
+
+use crate::{_macro_internal::ValueList, frame::response::result::ColumnSpec};
+
+pub struct RowSerializationContext<'a> {
+    columns: &'a [ColumnSpec],
+}
+
+impl<'a> RowSerializationContext<'a> {
+    pub fn column_by_idx(&self, i: usize) -> Option<&ColumnSpec> {
+        self.columns.get(i)
+    }
```

Comment:
I'm not exactly a fan of this... This doesn't allow to easily get information about the number of columns, and makes iteration over the columns slightly awkward. How about exposing this field through a getter that returns `&'a [ColumnSpec]`?